src/declarative-stack.ts

```diff
@@ -20,19 +21,18 @@ export class DeclarativeStack extends cdk.Stack {
     });
 
     const typeSystem = props.typeSystem;
-    const template = props.template;
+    const template = new TypedTemplate(props.template, { typeSystem });
 
     const context = new EvaluationContext({
       stack: this,
       template,
       typeSystem,
     });
+    const ev = new Evaluator(context);
 
-    new Evaluator(context).evaluateTemplate((fn) =>
-      // Changing working directory if needed, such that relative paths in the template are resolved relative to the
-      // template's location, and not to the current process' CWD.
-      _cwd(props.workingDirectory, fn)
-    );
+    _cwd(props.workingDirectory, () => {
+      ev.evaluateTemplate();
+    });
```

Comment:
The previous implementation changed the cwd per resource. There's no point to that really. This is better.